app/controllers/points_controller.rb

```diff
@@ -1,18 +1,23 @@
 # Defines actions for Points Controller
 class PointsController < ApplicationController
+  def index
+    @points = current_user.points
+  end
+
   def create
-    @point = Point.create(point_params)
-    if @point.save
-      redirect_to admin_points_path
-    else
-      flash.notice = 'You must select a point value.'
-      redirect_to new_admin_point_path
-    end
+    Point.create(point_params)
+    UserReward.create(user_reward_params)
```

Comment:
@iandouglas, @alischlereth

The way I have this set up is that when a user clicks on the "Redeem points for this reward" link, it creates a new point with the negative value of the Reward's cost in order to preserve the history of all points gained and spent, but it also creates a UserReward in order to show that the user now has this reward. I'm afraid that this doesn't adhere to SRP since this UserReward is being created in the PointsController, but I don't know how else to accomplish these two things with a single click.  Any thoughts?

Reply:
You could potentially create a RedeemController and clicking the link goes to a create action.